### PR TITLE
Prevent a double save on the post and filter before save instead

### PR DESCRIPTION
**Pull request by jaredcobb**
The previous functionality was causing the post to save twice. This will accomplish the same goal (replace the `lazyloaded` class with `lazyload`) but will do it as a filter before the post is saved.

### CODE REVIEW

File: php/class-photonfill.php [hunk 1]
```diff
@@ -779,17 +779,12 @@ public function get_breakpoint_url( $attachment_id, $img_size, $breakpoint, $pix
...
 		 */
-		public function swap_lazyload_classes( $post_id ) {
-			$current_post = get_post( $post_id );
-			$current_post->post_content = preg_replace( '/(class="[^"]*)(lazyloaded)([^"]*")/i', '$1' . 'lazyload' . '$3', $current_post->post_content );
-
-			remove_action( 'save_post', array( $this, 'swap_lazyload_classes' ) );
-			wp_update_post( $current_post );
-			add_action( 'save_post', array( $this, 'swap_lazyload_classes' ) );
+		public function swap_lazyload_classes( $content ) {
+			return preg_replace( '/(class=\\\\"[^"]*)(lazyloaded)([^"]*")/i', '$1' . 'lazyload' . '$3', $content );
```
Comment (mslinnea) [inline php/class-photonfill.php]: Why did you add `\\\\` ?  Just curious.

**Comment by mslinnea** [comment]
🍨

File: php/class-photonfill.php (hunk 1 above)
Comment (jaredcobb) [inline php/class-photonfill.php]: the content contains _escaped_ text at this stage... (so double quotes are being escaped with a `\` at this point in preparation for mysql).

in a regex you need to escape a backslash with another literal backslash (because a backslash is an escape character).

and even more confusing, `preg_replace` will escape the escape characters in the string before evaluating the replacement http://stackoverflow.com/questions/2222643/php-preg-replace

so `\\\\` gets translated to `\\` by `preg_replace` which gets translated to `\` by the regex. 😄 

> regex is really great to work with
- said no one ever

Comment (mslinnea) [inline php/class-photonfill.php]: Wow, could they make it any more complicated??  Probably.  Thanks for the explanation.  Now I know!
